plugins/uefi-capsule/fu-uefi-capsule-plugin.c

```diff
@@ -874,18 +876,33 @@ fu_uefi_capsule_plugin_check_cod_support(FuPlugin *plugin, GError **error)
 		return FALSE;
 	}
 
-#ifndef __aarch64__
-	/* several models with Insyde firmware have been released where OsIndications advertises
-	 * support for CoD, but it simply does not work */
-	if (!fu_context_has_hwid_flag(ctx, "uefi-allow-cod")) {
+	/* no table, nothing to check */
+	path = fu_path_from_kind(FU_PATH_KIND_ACPI_TABLES);
+	fn = g_build_filename(path, "UEFI", NULL);
+	if (!g_file_test(fn, G_FILE_TEST_EXISTS))
+		return TRUE;
+
+	/* if we have a table, parse it and validate it*/
+	blob = fu_bytes_get_contents(fn, error);
+	if (blob == NULL)
+		return FALSE;
+
+	acpi_uefi = fu_acpi_uefi_new(blob, error);
+	if (acpi_uefi == NULL)
+		return FALSE;
+
+	if (!fu_acpi_uefi_is_insyde(acpi_uefi)) {
+		g_debug("IBV is not Insyde");
+		return TRUE;
+	}
+
+	if (!fu_acpi_uefi_working_cod(acpi_uefi)) {
 		g_set_error_literal(error,
 				    G_IO_ERROR,
 				    G_IO_ERROR_NOT_SUPPORTED,
-				    "Capsule-on-Disk is supported, but non-aarch64 platforms "
-				    "require the 'uefi-allow-cod' quirk for it to be used");
+				    "Capsule-on-Disk may have a firmware bug");
```

Comment:
should we leave uefi-allow-cod or remove it? I guess it might be useful for other vendors.

Reply:
I would rather remove it unless we find another bug.